control/freqplot.py

```diff
@@ -838,6 +861,8 @@ def nyquist_plot(syslist, omega=None, plot=True, omega_limits=None,
         ax.set_xlabel("Real axis")
         ax.set_ylabel("Imaginary axis")
         ax.grid(color="lightgray")
+        ax.set_xlim(-maximum_magnitude*1.1, maximum_magnitude*1.1)
+        ax.set_ylim(-maximum_magnitude*1.1, maximum_magnitude*1.1)
```

Comment:
The plot limits are really tied to the `maximum_magnitude` now. Even for systems like this:

```python
from control import TransferFunction, nyquist_plot
G = TransferFunction([1],[500, 500, 0])
nyquist_plot(G)
```

| Before | After |
| --------- | ------ |
| ![Figure_1_before](https://user-images.githubusercontent.com/4623504/142771989-2ec3a0b2-8263-447a-8c45-293cfdc5a496.png) | ![Figure_1](https://user-images.githubusercontent.com/4623504/142771941-834acdda-76b1-4b35-b1ed-d3c9726d6b81.png) |



Is that desired?

Reply:
I think (though I am not positive) that @bnavigator's system might be a pathological case because you are mostly concerned with systems cascaded with their controllers, in which case you won't have a very low loop gain as this system does. 

That said, I rewrote the indentation code to add a consistent number of points at every indent, and further reduced the indent radius, which make it so everything but the most extremely low-gain systems have large contours that go outside of the plot window. This also helps with other pathological cases. 

also: 
* removed an indent test that confirms that no indentation happens when first omega is too large  
* made the red lines have the same line style as their associated contour. 
Updated after plots: 
<img width="418" alt="image" src="https://user-images.githubusercontent.com/58706249/142932954-7b242fde-9af4-4ebe-b026-a8f8eba44889.png">
<img width="394" alt="image" src="https://user-images.githubusercontent.com/58706249/142933005-a086f4af-23cb-492a-9c71-9cb952a50151.png">
And @bnavigator's example: 
<img width="510" alt="image" src="https://user-images.githubusercontent.com/58706249/142933140-8d558478-d0bf-42cd-ab51-cfcff4a688e8.png">

System with poles not at origin: 
Before: 
<img width="409" alt="image" src="https://user-images.githubusercontent.com/58706249/142934496-5afe47d8-23fc-457d-bbce-b394cb6f8571.png">

After: 
<img width="532" alt="image" src="https://user-images.githubusercontent.com/58706249/142934544-f1e22cc9-000b-4cfd-bfb5-812433c2c3ff.png">

Reply:
Okay, bad example.

Let's consider a system without undamped resonance:

```python
In [1]: from control import TransferFunction, nyquist_plot
   ...: G = TransferFunction([0.5],[1, 1, 1])
   ...: nyquist_plot(G)
```

| before | after |
| --- | ---|
| ![nyquist_before](https://user-images.githubusercontent.com/4623504/143132987-60924968-0840-49b0-8832-a50a8b91a242.png) | ![nyquist_after](https://user-images.githubusercontent.com/4623504/143132992-7d611fbd-1740-43e3-80db-808a0e87a386.png) |

Reply:
Do you think this would be resolved by adding a test for whether there was any `maximum_magintude` contour drawn, and if there wasn't, then to let it zoom the plot by the default amount?

Reply:
Sounds reasonable.